### PR TITLE
Add discoveryserver helm chart

### PR DESCRIPTION
**Pull request by eduartua**
The discoveryserver deployment installed by this chart has been tested with the `hack/insta-discovery bring-up` script and it worked as expected.

### CODE REVIEW

File: discoveryserver.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ spec:
         fsGroup: 9001
       containers:
       - name: discoveryserver
-        image: quay.io/etcd/discoveryserver:latest
+        image: quay.io/etcd/discoveryserver
```
Comment (victortrac) [inline discoveryserver.yaml]: Why are you making this change?

File: kubernetes/helm/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+apiVersion: apps/v1
```
Comment (victortrac) [inline kubernetes/helm/deployment.yaml]: What is this file doing here?

File: kubernetes/helm/discoveryserver/dev.values.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  runAsUser: 9001
+  runAsGroup: 9001
+  fsGroup: 9001
+
+securityContext:
+  allowPrivilegeEscalation: false
+
+environment:
+  - name: DISC_ETCD
+    value: "http://etcd-cluster-client.etcd-operator.svc.cluster.local:2379"
```
Comment (victortrac) [inline kubernetes/helm/discoveryserver/dev.values.yaml]: Does the etcd-operator put the etcd cluster in the operator's namespace?  This seems wrong.  Shouldn't the etcd clusters live in a separate namespace from the operator?

File: discoveryserver.yaml (hunk 1 above)
Comment (eduartua) [reply discoveryserver.yaml]: This should not have changed.

File: kubernetes/helm/deployment.yaml (hunk 2 above)
Comment (eduartua) [reply kubernetes/helm/deployment.yaml]: This file shouldn't be there. Removed.

File: kubernetes/helm/discoveryserver/dev.values.yaml (hunk 3 above)
Comment (eduartua) [reply kubernetes/helm/discoveryserver/dev.values.yaml]: The default is that the operator is going to manage cluster in the same namespace.
From etcd-operator docs:
_Default etcd operator behavior is to only manage etcd clusters created in the same namespace. It is possible to deploy an etcd operator with special option to manage clusterwide etcd clusters._

Comment (victortrac) [reply kubernetes/helm/discoveryserver/dev.values.yaml]: @eduartua A couple of nitpicks, then. If we want to use a namespaced etcd cluster operator, and we want to keep the etcd cluster within the namespace as the discoveryserver, then we should put the etcd-operator, etcd and discovery server into the namespace, like a namespace called `discoveryserver`. The etcd cluster should be named something descriptive to its purpose, like `discoveryserver-etcd`.

`etcd-cluster-client` is a bad name.

Comment (eduartua) [reply kubernetes/helm/discoveryserver/dev.values.yaml]: I do agree. I've made the changes to have them all running in the same namespace. Etcd cluster name has been changes to `discovery-etcd-cluster`

File: kubernetes/helm/discoveryserver/dev.values.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        - dev.discovery.etcd.io
+        - www.dev.discovery.etcd.io
+
+resources:
+   limits:
+     cpu: 200m
+     memory: 1200Mi
+   requests:
+     cpu: 100m
+     memory: 128Mi
```
[the remark this answers is not in the document]
Comment (eduartua) [reply kubernetes/helm/discoveryserver/dev.values.yaml]: That configuration is wrong. I think I mixed those with etcd-cluster values. The steady-state value for discoveryserver is around 20 Mi.

File: kubernetes/helm/etcd-operator/dev.values.yaml [hunk 5]
```diff
@@ -115,8 +115,8 @@ etcdCluster:
       fsGroup: 9000
     resources:
       limits:
-        cpu: 100m
-        memory: 128Mi
+        cpu: 200m
+        memory: 1200Mi
       requests:
         cpu: 100m
         memory: 128Mi
```
[the remark this answers is not in the document]
Comment (eduartua) [reply kubernetes/helm/etcd-operator/dev.values.yaml]: Changed to request 600Mi as the steady-state is around 500Mi.

**Pull request merged**